### PR TITLE
[API][Behat] Cover scenarios for applying correct taxes

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | master          |
| Bug fix?        | no                                                       |
| New feature?    | yes                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no |
| Related tickets |                       |
| License         | MIT                                                          |

### CODE REVIEW

File: features/taxation/applying_taxes/applying_correct_taxes_for_items_with_the_same_tax_rate.feature [hunk 1]
```diff
@@ -40,10 +40,10 @@ Feature: Apply correct taxes for items with the same tax rate
         Then my cart total should be "$73.80"
         And my cart taxes should be "$13.80"
 
-    @ui
+    @ui @api
     Scenario: Proper taxes after changing item quantity
         Given I have 3 products "PHP T-Shirt" in the cart
         And I have 2 products "Symfony Hat" in the cart
-        When I change "PHP T-Shirt" quantity to 1
+        When I change product "PHP T-Shirt" quantity to 1 in my cart
```
Comment (TheMilek) [inline features/taxation/applying_taxes/applying_correct_taxes_for_items_with_the_same_tax_rate.feature:47]:
```suggestion
        When I change the number of "PHP T-Shirt" products in the cart to 1
```
I know we have to clarify, that we are mentioning quantity here, but still what do you think?

**Review by TheMilek — APPROVED** [review]

File: features/taxation/applying_taxes/applying_correct_taxes_for_items_with_the_same_tax_rate.feature (hunk 1 above)
Comment (GSadee) [reply features/taxation/applying_taxes/applying_correct_taxes_for_items_with_the_same_tax_rate.feature:47]: I'm not convinced as we are using `quantity` keyword both in API and UI for the number of products in the cart, and also in Behat scenarios, so it is clear enough IMO 😃

**Review by AdamKasp — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/CommandHandler/Cart/PickupCartHandler.php [hunk 2]
```diff
@@ -108,4 +110,15 @@ private function getLocaleCode(?string $localeCode, ChannelInterface $channel):
...
+
+    private function getDefaultAddress(CustomerInterface $customer): ?AddressInterface
+    {
+        $defaultAddress = $customer->getDefaultAddress();
+        if (null !== $defaultAddress) {
+            $clonedAddress = clone $defaultAddress;
+            $clonedAddress->setCustomer(null);
+        }
+
+        return $clonedAddress ?? null;
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/CommandHandler/Cart/PickupCartHandler.php:122]: We could reduce one "if" statement :P 
```suggestion
        $clonedAddress = null;
        
        if (null !== $defaultAddress) {
            $clonedAddress = clone $defaultAddress;
            $clonedAddress->setCustomer(null);
        }

        return $clonedAddress;
```
or

```suggestion
        if (null !== $defaultAddress) {
            $clonedAddress = clone $defaultAddress;
            $clonedAddress->setCustomer(null);
            
            return $clonedAddress;
        }

        return null;
```

File: src/Sylius/Bundle/ApiBundle/spec/CommandHandler/Cart/PickupCartHandlerSpec.php [hunk 3]
```diff
@@ -77,6 +79,7 @@ function it_picks_up_a_new_cart_for_logged_in_shop_user(
         $channel->getDefaultLocale()->willReturn($locale);
 
         $customerRepository->findOneBy(['email' => 'sample@email.com'])->willReturn($customer);
+        $customer->getDefaultAddress()->willReturn($address);
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/spec/CommandHandler/Cart/PickupCartHandlerSpec.php:82]: that's funny, that this works, as we are cloning it :P 

Anyway, we should cover the null case well

**Review by lchrusciel — APPROVED** [review]

**Comment by lchrusciel** [comment]
Thanks, Grzegorz! :1st_place_medal:

**Pull request merged**